src/nunit.analyzers/EqualToIncompatibleTypes/EqualToIncompatibleTypesAnalyzer.cs

```diff
@@ -25,46 +25,68 @@ public class EqualToIncompatibleTypesAnalyzer : BaseAssertionAnalyzer
 
         protected override void AnalyzeAssertInvocation(OperationAnalysisContext context, IInvocationOperation assertOperation)
         {
-            if (!AssertHelper.TryGetActualAndConstraintOperations(assertOperation,
-                out var actualOperation, out var constraintExpression))
+            IOperation? actualOperation;
+            IOperation? expectedOperation;
+
+            if (assertOperation.TargetMethod.Name.Equals(NunitFrameworkConstants.NameOfAssertAreEqual) ||
+                assertOperation.TargetMethod.Name.Equals(NunitFrameworkConstants.NameOfAssertAreNotEqual))
             {
-                return;
-            }
+                actualOperation = assertOperation.GetArgumentOperation(NunitFrameworkConstants.NameOfActualParameter);
+                expectedOperation = assertOperation.GetArgumentOperation(NunitFrameworkConstants.NameOfExpectedParameter);
 
-            foreach (var constraintPartExpression in constraintExpression.ConstraintParts)
+                CheckActualVsExpectedOperation(context, actualOperation, expectedOperation);
+            }
+            else
             {
-                if (HasIncompatiblePrefixes(constraintPartExpression)
-                    || HasCustomEqualityComparer(constraintPartExpression)
-                    || constraintPartExpression.HasUnknownExpressions())
+                if (!AssertHelper.TryGetActualAndConstraintOperations(assertOperation,
+                    out actualOperation, out var constraintExpression))
                 {
                     return;
                 }
 
-                var constraintMethod = constraintPartExpression.GetConstraintMethod();
-
-                if (constraintMethod?.Name != NunitFrameworkConstants.NameOfIsEqualTo
-                    || constraintMethod.ReturnType?.GetFullMetadataName() != NunitFrameworkConstants.FullNameOfEqualToConstraint)
+                foreach (var constraintPartExpression in constraintExpression.ConstraintParts)
                 {
-                    continue;
-                }
+                    if (HasIncompatiblePrefixes(constraintPartExpression)
+                        || HasCustomEqualityComparer(constraintPartExpression)
+                        || constraintPartExpression.HasUnknownExpressions())
+                    {
+                        return;
```

Comment:
With what I know now, I would say yes but this was explicitly changed from a `continue` into a `return` by @Dreamescaper 4 months ago (commit 1a004f7454250bbd01b0433d0605b9a1d9999623)
Changing this fails the following test:
```
var actual = new[] { 1, 2, 2, 1 };
Assert.That(actual, Has.All.EqualTo(1).Or.EqualTo(2));
```
This results in a `Has.AllEqualTo(1)` part which would be ignored and a `EqualTo(2)` part which is flagged as an error.
It seems that NUnit groups this to 'Has.All(.EqualTo(1).Or.EqualTo(2))

I tried to add a test that would miss detecting an error like in:
`Assert.That(5, Is.EqualTo("5").Using(new AsInt32Comparer()) & Is.EqualTo("5"));`
But I'm having trouble getting the NUnit to use a comparer:
```
            IEqualityComparer comparer = new AsInt32Comparer();
            Assert.That(comparer.Equals("5", 5), Is.True, "Comparer(string, int)");
            Assert.That(comparer.Equals(5, "5"), Is.True, "Comparer(int, string)");

            Assert.That("5", Is.EqualTo(5).Using(comparer), "NUnit using Comparer(string, int)");
            Assert.That(5, Is.EqualTo("5").Using(comparer), "NUnit using Comparer(int, string)");

            Assert.That(5, Is.EqualTo("5").Using(comparer) & Is.EqualTo(5), "NUnit Combined");
```
The first two asserts pass, proofing that my comparer works. The NUnit versions fail. This looks like the almost 4 year old bug: https://github.com/nunit/nunit/issues/1897
```
        public virtual bool CanCompare(object x, object y)
        {
            if (x is string && y is string)
                return true;

            if (x is IEnumerable || y is IEnumerable)
                return false;

            return true;
        }
```
Because only one of the arguments is `string` and `string` implements `IEnumerable` the method always returns `false`.

Reply:
Yeah, it is due to https://github.com/nunit/nunit.analyzers/issues/218.
Constraint expression is split by .And property, but it doesn't account for .All property operator in first part, which affects both parts.